### PR TITLE
Fix synopsysctl create native

### PR DESCRIPTION
**Pull request by jdartigalongue**
/assign @rrati @msenmurugan @mphammer

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#" title="Author self-approved">jdartigalongue</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#" title="Author self-approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#pullrequestreview-265087420" title="Approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/api/types.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ package api
 import (
 	"github.com/blackducksoftware/horizon/pkg/components"
 	routev1 "github.com/openshift/api/route/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (msenmurugan) [inline pkg/api/types.go]: rename to metav1 will make consistent

File: pkg/apps/blackduck/blackduck.go [hunk 2]
```diff
@@ -1910,25 +1910,30 @@ func (b *Blackduck) Ensure(blackduck *v1.Blackduck) error {
...
+	}
+
+	cp, err := store.GetComponents(version, b.config, b.kubeClient, bd)
+	if err != nil {
+		return nil, err
+	}
+
+	switch strings.ToUpper(compType) {
+	case CRDResources:
+		return cp.Filter("component notin (postgres)")
```
Comment (msenmurugan) [inline pkg/apps/blackduck/blackduck.go]: @jdartigalongue Is there a way to use CRUD updater instead of maintaining the new method?

Comment (jdartigalongue) [reply pkg/apps/blackduck/blackduck.go]: The CRUD updater doesn't return kube objects and would be overkill for this. We just want to filter the components by labels to retrieve what's requested.

File: pkg/apps/blackduck/blackduck.go [hunk 3]
```diff
@@ -1910,25 +1910,30 @@ func (b *Blackduck) Ensure(blackduck *v1.Blackduck) error {
...
-	//return nil, fmt.Errorf("invalid components type '%s'", compType)
-	return nil, nil
+	//If the version is not specified then we set it to be the latest.
+	if err := b.ensureVersion(bd); err != nil {
+		return nil, err
+	}
+
+	version, ok := publicVersions[bd.Spec.Version]
+	if !ok {
+		return nil, fmt.Errorf("versiom %s is not supported", bd.Spec.Version)
```
Comment (rrati) [inline pkg/apps/blackduck/blackduck.go]: versiom -> version

File: pkg/apps/store/store.go [hunk 4]
```diff
@@ -33,27 +33,27 @@ func Register(name types.ComponentName, function interface{}) {
...
+		//fmt.Fprintln(os.Stderr, "loaded configmap: "+name)
 	case types.PvcCreater:
 		ComponentStore.PVC[name] = function.(types.PvcCreater)
 	case func(*protoform.Config, *kubernetes.Clientset, *v1.Blackduck) types.SecretInterface:
 		if ComponentStore.Secret == nil {
 			ComponentStore.Secret = make(map[types.ComponentName]types.SecretCreater)
 		}
 		ComponentStore.Secret[name] = function.(func(*protoform.Config, *kubernetes.Clientset, *v1.Blackduck) types.SecretInterface)
-		fmt.Println("loaded secret: " + name)
+		//fmt.Fprintln(os.Stderr, "loaded secret: "+name)
```
Comment (rrati) [inline pkg/apps/store/store.go]: Remove commented out code

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#" title="Author self-approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#pullrequestreview-265087420" title="Approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#pullrequestreview-265096721" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#" title="Author self-approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#issuecomment-514252014" title="LGTM">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/701#pullrequestreview-265096721" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**